File: clients/android/trace/app/src/main/java/com/example/brandon/trace/Message.java
package com.example.brandon.trace;

/**
 * Message class used to send & receive information via a ControlConnection.
 */
public class Message {
    public static final String NEW = "new";
    public static final String LIST = "list";
    public static final String DONE = "done";

    public String Type;
    public String File;
    public int Length;
    public String Body;

    public Message(String Type, String File, int Length, String Body) {
        this.Type = Type;
        this.File = File;
        this.Length = Length;
        this.Body = Body;
    }
}

File: lib/message.go
package lib

const (
	// New - used when sending a new file
	New = "new"

	// List - used when sending the file as part of a list
	List = "list"

	// Done - used when there are no more file parts to send
	Done = "done"
)

// Message - Used to contain information sent over the wire
type Message struct {
	Type   string
	File   string
	Length int
	Body   string
}
